product-service/src/libs/utils.ts

```diff
@@ -0,0 +1,15 @@
+export const findItemById = <T extends { id: string }>(
+  list: T[],
+  id: string
+): T | undefined => {
+  let found;
```

Comment:
Quite interesting)) Maybe we can use just list.find(item => item.id === id) ? )

Reply:
yeah, that was done deliberately, just to move the business logic out of lamdas inspired by separation of concerns. Inside the `findItemById` function, I agree, that can be simplified with a more functional way of items iteration.